pkg/endpoint/endpoint.go

```diff
@@ -1028,6 +1027,13 @@ func (e *Endpoint) RegenerateWait(reason string) error {
 	return nil
 }
 
+// GetContainerName returns the name of the container for the endpoint.
+func (e *Endpoint) GetContainerName() string {
```

Comment:
We seem a little inconsistent in whether we lock when reading this field (we lock for invocations of endpointmanager.Remove except for tests at pkg/endpointmanager/manager_test.go). I'm not sure that it's worth fixing, since it seems to mostly be read in tests and it should be write-once.
The set function does lock, however... so yeah.

Reply:
Yeah, that's one of the reasons why I'm trying to refactor to hide direct access to fields; we need consistency in the conditions under which fields are accessed, and should not exposes mutexes as a result.